pkg/kubectl/custom_column_printer.go

```diff
@@ -35,6 +37,95 @@ const (
 	flags             = 0
 )
 
+// MassageJSONPath attempts to be flexible with JSONPath expressions, it accepts:
+//   * metadata.name (no leading '.' or curly brances '{...}'
+//   * {metadata.name} (no leading '.')
+//   * .metadata.name (no curly braces '{...}')
+//   * {.metadata.name} (complete expression)
+// And transforms them all into a valid jsonpat expression:
+//   {.metadata.name}
+func massageJSONPath(pathExpression string) string {
+	if len(pathExpression) == 0 {
+		return pathExpression
+	}
+	if pathExpression[0] != '{' {
+		if pathExpression[0] == '.' {
+			return fmt.Sprintf("{%s}", pathExpression)
+		} else {
+			return fmt.Sprintf("{.%s}", pathExpression)
+		}
+	} else {
+		if pathExpression[1] == '.' {
+			return pathExpression
+		} else {
+			return fmt.Sprintf("{.%s}", pathExpression[1:len(pathExpression)-1])
+		}
+	}
+}
+
+// NewCustomColumnsPrinterFromSpec creates a custom columns printer from a comma separated list of <header>:<jsonpath-field-spec> pairs.
+// e.g. NAME:metadata.name,API_VERSION:apiVersion creates a printer that prints:
+//
+//      NAME               API_VERSION
+//      foo                bar
+func NewCustomColumnsPrinterFromSpec(spec string) (*CustomColumnsPrinter, error) {
+	parts := strings.Split(spec, ",")
+	if len(parts) == 0 {
+		return nil, fmt.Errorf("custom-columns format specified but no custom columns given")
```

Comment:
I think `len(parts)` will be `1` even when `spec == ""`. 
http://play.golang.org/p/hSUyTWnMzt